app/controllers/users_controller.rb

```diff
@@ -0,0 +1,105 @@
+class UsersController < ApplicationController
+  before_action :require_login, only: [:upvote]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    return head :not_found if @user.nil?
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "#{@user.title} was successfully added!"
+      redirect_to User_path(@user.id)
```

Comment:
This should be lower case here.  I'm actually surprised Rails is chill about the casing here.

```suggestion
      redirect_to user_path(@user.id)
```